sys/net/gnrc/netif/init_devs/include/init_devs.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_auto_init_gnrc_netif
+ * @{
+ *
+ * @file
+ * @brief       common netif device initialization definitions
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ */
+
+#ifndef INIT_DEVS_H
+#define INIT_DEVS_H
+
+#include "thread.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   extra stack size if ieee802154 security is enabled
+ */
+#define IEEE802154_SECURITY_EXTRA_STACKSIZE (32)
```

Comment:
I'm afraid that was not enough.
Seems like fragmented packets are really taxing the stack.

When I did `ping fe80::e4ea:aff8:af5d:eae5%6 -s 1200` I got

```
2020-11-26 22:01:46,610 # 	  6 | at86rf215 [sub GHz]  | bl rx    _ |   2 |   2048 ( 1112) (  936) | 0x20001208 | 0x20001844 
```

(I changed the extra stacksize to 1024 to see how big it gets)

I'd say

```suggestion
#define IEEE802154_SECURITY_EXTRA_STACKSIZE (128)
```

will be enough

Reply:
Something else is odd: when I do send fragmented packets a couple of times, I eventually get

```
2020-11-26 22:05:16,947 #  ping fe80::e4ea:aff8:af5d:eae5%6 -s 1200
2020-11-26 22:05:16,948 # error: packet buffer full
2020-11-26 22:05:17,939 # error: packet buffer full
2020-11-26 22:05:18,947 # error: packet buffer full
```

I don't see this without security.

Reply:
I hope that is fixed now. The `gnrc_pktsnip` of the MIC was not release before.

Reply:
Do I still need to increase the stack, or has this fixed the other issue too?

Reply:
> Do I still need to increase the stack, or has this fixed the other issue too?

32 byte extra is still too little on RX

```
2020-11-27 23:28:27,852 # 	  5 | udp                  | bl rx    _ |   5 |   1024 (  248) (  776) | 0x20004a94 | 0x20004d9c 
2020-11-27 23:28:27,853 # 	  6 | at86rf215 [sub GHz]  | running  Q |   2 |   1056 ( 1056) (    0) | 0x20001208 | 0x200014ec 
2020-11-27 23:28:27,854 # 	  7 | at86rf215 [2.4 GHz]  | bl rx    _ |   2 |   1056 (  932) (  124) | 0x20001628 | 0x2000190c 
2020-11-27 23:28:27,855 # 	    | SUM                  |            |     |   8768 ( 4124) ( 4644)
2020-11-27 23:28:27,855 # 
2020-11-27 23:28:27,855 # *** halted.
2020-11-27 23:28:27,855 # 
2020-11-27 23:28:27,855 # Inside isr -12
```

with `at86rf215` I see a maximum stack use of 1112 bytes with security enabled and receiving fragmented packets, so `IEEE802154_SECURITY_EXTRA_STACKSIZE` would be 88, maybe round up to 96 or 128?

But the good news it that fragmented packets now work reliably too!